### PR TITLE
FEATURE: Switch InfluxDB sources per chart in Dashboards

### PR DESCRIPTION
**Pull request by 121watts**
- [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1088 

### The problem
Users want to be able use more than one data source per dashboard. 

### The Solution
Add the option to switch between any source on any chart in a dashboard.

### Rough feel of the UX
source selector

### CODE REVIEW

File: ui/src/dashboards/components/DisplayOptions.js [hunk 1]
```diff
@@ -2,6 +2,9 @@ import React, {Component, PropTypes} from 'react'
 
 import GraphTypeSelector from 'src/dashboards/components/GraphTypeSelector'
 import AxesOptions from 'src/dashboards/components/AxesOptions'
+import SourceSelector from 'src/dashboards/components/SourceSelector'
+
+import _ from 'lodash'
```
Comment (jaredscheib) [inline ui/src/dashboards/components/DisplayOptions.js]: extremely minor, but generally we've been putting these libraries just under the React import itself, before component imports

**Comment by 121watts** [comment]
### TODO
- [ ] make `DisplayOptions` responsive
- [ ] add query source indicator to dashboard cell

File: server/dashboards_test.go [hunk 2]
```diff
@@ -236,6 +237,7 @@ func Test_newDashboardResponse(t *testing.T) {
 						H:  0,
 						Queries: []chronograf.DashboardQuery{
 							{
+								Source:  "/chronograf/v1/sources/2",
```
[the remark this answers is not in the document]
Comment (goller) [reply server/dashboards_test.go]: @jaredscheib @121watts 

Currently, the `source` will either have a URL or will be an empty string.  Regarding if the test should test for empty string, you bet.

[the remark this answers is not in the document]
Comment (goller) [reply server/dashboards_test.go]: Updated in 7f60777f806b114bfe256ea51a2e154ecab983a9

[the remark this answers is not in the document]
Comment (jaredscheib) [reply server/dashboards_test.go]: Word

**Comment by 121watts** [comment]
@jaredscheib after talking it through with @alexpaxton, we think the best spot for the `SourceSelector` is in `QueryControls` aka right above the query tabs.  This solves all the above UI/UX issues and we think it looks / feels a lot better.

**Comment by 121watts** [comment]
@jaredscheib now, we can take advantage of the query error handling:

error handling

**Comment by jaredscheib** [comment]
Looks good to go for merge on my end, other than a changelog update and I think it'd be good for @121watts to review the latest bit of work from me and @alexpaxton.

The latest commits entail the following changes: 1) use SourceIndicator component (i.e. an icon instead of the host name label) on dashboard cells; 2) refactor SourceIndicator component in service of that modification; 3) add the label `Source:` next to the SourceSelector's new position.

**Comment by jaredscheib** [comment]
Actually, product decision on SourceIndicator is that we're not going to include an indication of alternate source in Cell Headers.

**Pull request merged**